src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -410,8 +435,8 @@ internal static async Task CheckNonOptionalSettings(IEnumerable<KeyValuePair<str
                                 else if (!secrets.Any(v => v.Key.Equals(appSettingName, StringComparison.OrdinalIgnoreCase)))
                                 {
                                     ColoredConsole
-                                        .WriteLine(WarningColor($"Warning: Cannot find value named '{appSettingName}' in {SecretsManager.AppSettingsFileName} that matches '{token.Key}' property set on '{binding["type"]?.ToString()}' in '{filePath}'. " +
-                                            $"You can run 'func azure functionapp fetch-app-settings <functionAppName>' or specify a connection string in {SecretsManager.AppSettingsFileName}."));
+                                        .WriteLine(WarningColor($"Warning: Cannot find value named '{appSettingName}' in {orConfigSources} that matches '{token.Key}' property set on '{binding["type"]?.ToString()}' in '{filePath}'. " +
```

Comment:
Same here. nit: move exception message string to constants. This will avoid defining variables orConfigSources / andConfigSources